airflow/providers/google/cloud/transfers/sftp_to_gcs.py

```diff
@@ -130,18 +140,21 @@ def execute(self, context: Context):
 
             for file in files:
                 destination_path = file.replace(base_path, self.destination_path, 1)
-                self._copy_single_object(gcs_hook, sftp_hook, file, destination_path)
+                if self.use_stream:
+                    self._stream_single_object(sftp_hook, gcs_hook, file, destination_path)
+                else:
+                    self._copy_single_object(sftp_hook, gcs_hook, file, destination_path)
 
         else:
             destination_object = (
                 self.destination_path if self.destination_path else self.source_path.rsplit("/", 1)[1]
             )
-            self._copy_single_object(gcs_hook, sftp_hook, self.source_path, destination_object)
+            self._copy_single_object(sftp_hook, gcs_hook, self.source_path, destination_object)
```

Comment:
Streaming is missing here - Did you intentionally only add streaming in the case of wildcard in file path?